### DISCUSSION TITLE
Cargo watch / Compiling

### DISCUSSION THREAD

**Discussion by kamalkech** [Brendonovich/prisma-client-rust#162] [General]
The app take a looot of time to re-compile the app to test on mode dev  !!

**Comment by Brendonovich** [comment]
Yeah, it's a big library. Some people have been testing 0.6.2 and it's cut down compile times anywhere from 15 to 35%, so that may help you out. In your Cargo.tomls change `tag = "0.6.1"` to `branch = "0.6.2"` and check the [specifying your database](https://prisma-client-rust-git-062-brendonovich.vercel.app/getting-started/installation#specifying-your-database) section in the new docs.

**Reply by kamalkech** [reply]
ok i will check and i am using sqlite tag of prisma-client-rust and prisma-client-rust-cli ? 
to select the database as feature, in the documentation i read "The possible values are postgresql, mysql, sqlite, mssql and mongodb." but in your example in github i found sqlite-create-many !!

**Reply by Brendonovich** [reply]
> in your example in github i found sqlite-create-many !!

That's a separate feature documented [on this page](https://prisma.brendonovich.dev/writing-data/create)

**Comment by kamalkech** [comment]
I update all this and still compiling and building slow than typescript + prisma project !!

**Reply by oscartbeaumont** [reply]
Rust is going to be slower than a Typescript and that's just how it's always going to be. The number of checks Rust has to do to assert that your code is valid and safe takes time and Typescript can avoid a lot of the same sort of time consuming checks because of it's different architecture such as having a runtime (which does garbage collection and the like). That is a pretty crude explanation but I would recommend reading up more about the languages. Rust allows you to trade off compile-time performance for better runtime performance.

Prisma Client Rust has had especially bad compile times because of its inclusion of the Prisma Engines which are huge. This design decision has allows the project to get to where it is today and the recent work on the 0.6.2 branch has made this significantly better.

**Reply by kamalkech** [reply]
probably i will agree with that, thank you @oscartbeaumont

**Comment by Brendonovich** [comment]
Rust isn't TypeScript. It's fast at the expense of compiling slowly. You've got to choose what you value more.

**Reply by kamalkech** [reply]
i know rust is not typescript, just i am surprised that compiler is too slow
